src/main/java/com/moneylendingapp/services/impl/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,28 @@
+package com.moneylendingapp.services.impl;
+
+import com.moneylendingapp.entities.User;
+import com.moneylendingapp.entities.UsersDetails;
+import com.moneylendingapp.repositories.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+
+@Service
+@RequiredArgsConstructor
+public class UserDetailsServiceImpl implements UserDetailsService {
+
+    private final UserRepository userRepo;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+
+        Optional<User> user = userRepo.findByUsername(username);
+        user.orElseThrow(() -> new UsernameNotFoundException(String.format("Username: %s not found", username)));
```

Comment:
This can be ` User user = userRepo.findByUsername(username).orElseThrow(() -> new UsernameNotFoundException(String.format("Username: %s not found", username)))`